### PR TITLE
Type annotation improvements

### PR DESCRIPTION
**Pull request by nsoranzo**
(some from the CWL branch).

Also remove some unused methods/functions.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/job_metrics/formatting.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ class FormattedMetric(NamedTuple):
 class JobMetricFormatter:
     """Format job metric key-value pairs for human consumption in Web UI."""
 
-    def format(self, key: Any, value: Any) -> FormattedMetric:
+    def format(self, key: str, value: Any) -> FormattedMetric:
         return FormattedMetric(str(key), str(value))
```
Comment (davelopez) [inline lib/galaxy/job_metrics/formatting.py]:
```suggestion
        return FormattedMetric(key, str(value))
```

File: lib/galaxy/tools/wrappers.py [hunk 2]
```diff
@@ -642,9 +642,9 @@ def __init__(
         self.collection = collection
 
         elements = collection.elements
-        element_instances = {}
+        element_instances: Dict[str, Union[DatasetCollectionWrapper, DatasetFilenameWrapper]] = {}
```
Comment (davelopez) [inline lib/galaxy/tools/wrappers.py]: Maybe a type alias for `Union[DatasetCollectionWrapper, DatasetFilenameWrapper]` named `ElementWrapper` will increase readability and reduce duplication here.

Comment (nsoranzo) [reply lib/galaxy/tools/wrappers.py]: Thanks for the good suggestion! I have added the `DatasetCollectionElementWrapper` type alias.

**Review by davelopez — APPROVED** [review]
Thank you!

**Pull request merged**